File: ts/project/.sdk/src/cmp/csharp/fragment/EntityCreateOp.fragment.cs
// Create operation fragment (spliced into the entity class by
// EntityOperation_csharp; only the EJECT region is emitted).

namespace ProjectNameSdk.Entity;

public class EntyClassCreateOpFragment : EntyClass
{
    public EntyClassCreateOpFragment(ProjectNameSDK client) : base(client) { }

// EJECT-START
public override object? Create(Dictionary<string, object?>? reqdata,
    Dictionary<string, object?>? ctrl = null)
{
    var ctx = utility.MakeContext(new Dictionary<string, object?>
    {
        ["opname"] = "create",
        ["ctrl"] = ctrl,
        ["match"] = match,
        ["data"] = data,
        ["reqdata"] = reqdata,
    }, entctx);

    return RunOp(ctx, () =>
    {
        if (ctx.Result != null)
        {
            if (ctx.Result.Resdata != null)
            {
                data = Helpers.ToMapAny(
                    Voxgig.Struct.StructUtils.Clone(ctx.Result.Resdata))
                    ?? new Dictionary<string, object?>();
            }
        }
    });
}
// EJECT-END
}

File: ts/project/.sdk/src/cmp/swift/fragment/EntityCreateOp.fragment.swift
// Create operation fragment (spliced into the entity class by
// EntityOperation_swift; only the EJECT region is emitted).

import Foundation

final class EntyClassCreateOpFragment: EntyClass {
// EJECT-START
public override func create(_ reqdata: VMap?, _ ctrl: VMap?) throws -> Value {
  var ctxmap: [String: Any?] = ["opname": "create", "match": match, "data": data]
  if let ctrl = ctrl { ctxmap["ctrl"] = ctrl }
  if let reqdata = reqdata { ctxmap["reqdata"] = reqdata }
  let ctx = utility.makeContext(ctxmap, entctx)

  return try runOp(ctx) {
    if let result = ctx.result {
      if !isNil(result.resdata) {
        self.data = clone(result.resdata).asMap ?? VMap()
      }
    }
  }
}
// EJECT-END
}
